### ISSUE TITLE
Dedrm removal not working for some regional Titles in Amazon 

### ISSUE THREAD

**Issue by Tushd31** [apprenticeharper/DeDRM_tools#1096]
Hi ,
I Came across your thread .
Looking for a new issue facing .Some Regional Titles (Marathi ,Tamil ,Telugu )from Amazon india kindle store are not getting downloaded on Kindle for pc version 1.19 .Tried till 1.21 ,didn't worked .
Error code - You need to update your Kindle for pc with latest version .
For 1.23 and above it worked ,but format is kfx-azw3 for .Titles can be seen in Calibre but conversion fails stating "locked by drm "
Tried alternative way , but no option to send book to pc via "Send it to kindle" from Kindle web reader or Amazon account for these titles .For all English titles it worked and was able to send to pc .
Trying this for 5 days with every possible combination to work out ,but didn't worked .
Using calibre - Latest Version 4.16
Dedrm plugin 6.7 
Kfx plugin from mobileread which was updated recently .
My observation as this issue is for titles which are not available to send to pc via USB transfer (No option available for these titles )
Awaiting for your response .

**Comment by ElleKayEm** [comment]
If you cannot download a book with unmodified Kindle for PC 1.24 (no change to renderer-test), then you will probably have to wait until DeDRM can be updated to handle the newer DRM on KFX format.  If you can download it with 1.24 or earlier, post a log of adding it to calibre:

https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Comment by Tushd31** [comment]
Thank you so much for reply .
Will check and add that log here .
Thanks again for reverting back :)

**Comment by Tushd31** [comment]
> If you cannot download a book with unmodified Kindle for PC 1.24 (no change to renderer-test), then you will probably have to wait until DeDRM can be updated to handle the newer DRM on KFX format. If you can download it with 1.24 or earlier, post a log of adding it to calibre:
> 
> https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

hi , 
Here is the log that i received with version 1.23 for kindle pc and calibre 4.16
calibre Debug log
calibre 4.16  embedded-python: True is64bit: False
Windows-10-10.0.17134-SP0 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.17134')
Python 2.7.16+
Windows: ('10', '10.0.17134', 'SP0', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 31, 0) && DeDRM (6, 7, 0) && Package KFX (from KFX Input) (1, 31, 0) && Obok DeDRM (6, 7, 0) && Set KFX metadata (from KFX Output) (1, 41, 1) && KFX Output (1, 41, 1) && KFX metadata reader (from KFX Input) (1, 31, 0) && KFX Input (1, 31, 0)
calibre 4.16  embedded-python: True is64bit: False
Windows-10-10.0.17134-SP0 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.17134')
Python 2.7.16+
Windows: ('10', '10.0.17134', 'SP0', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 31, 0) && DeDRM (6, 7, 0) && Package KFX (from KFX Input) (1, 31, 0) && Obok DeDRM (6, 7, 0) && Set KFX metadata (from KFX Output) (1, 41, 1) && KFX Output (1, 41, 1) && KFX metadata reader (from KFX Input) (1, 31, 0) && KFX Input (1, 31, 0)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 100.861627907 x 100.55257732
Using calibre Qt style: True
[0.00] Starting up...
[0.02] Showing splash screen...
[2.36] splash screen shown
[2.36] Initializing db...
[2.39] db initialized
[2.39] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
[7.08] main UI initialized...
[7.08] Hiding splash screen
[15.27] splash screen hidden
[15.27] Started up in 15.27 seconds with 0 books
stdout+stderr from file dialog helper: ['', '']
piped data from file dialog helper: ['}\x96\xfb\xf9\x16X86v-\x0es\x9e\x90\x0f$_P\xd0`\xcf\xecMz\xc1\xcdF\xa9HN\xea\x1a', 'C:\\Users\\AUM\\Documents\\My Kindle Content\\B07G146556_EBOK\\B07G146556_EBOK.azw']
Gather KFX-ZIP (from KFX Input) 1.31.0: Importing C:\Users\AUM\Documents\My Kindle Content\B07G146556_EBOK\B07G146556_EBOK.azw
Gather KFX-ZIP (from KFX Input): Gathered 4 file(s) as C:\Users\AUM\AppData\Local\Temp\calibre_7hyy3d\zxwua4.kfx-zip
DeDRM v6.7.0: Trying to decrypt zxwua4.kfx-zip
Using Library AlfCrypto DLL/DYLIB/SO
Decrypting KFX-ZIP ebook: zxwua4
Got DSN key from database default_key_3
Found 3 keys to try after 0.4 seconds
Decrypting KFX DRM voucher: amzn1.drm-voucher.v1.d6b6caba-0349-42ff-b1aa-fda6a96dcd3d.voucher
KFX DRM voucher successfully decrypted
DeDRM v6.7.0: Failed to decrypt with error: This book is licensed as KindleUnlimited. These tools are intended for use on purchased books.
DeDRM v6.7.0: Looking for new default Kindle Key after 0.4 seconds
searching for kinfoFiles in C:\Users\AUM\AppData\Local
Found K4PC 1.25+ kinf2018 file: C:\Users\AUM\AppData\Local\Amazon\Kindle\storage\.kinf2018
Found K4PC 1.9+ kinf2011 file: C:\Users\AUM\AppData\Local\Amazon\Kindle\storage\.kinf2011
Decrypted key file using IDString '2115683436' and UserName '41554d'
Decrypted key file using IDString '2115683436' and UserName '41554d'
DeDRM v6.7.0: Found 2 new keys
Decrypting KFX-ZIP ebook: zxwua4
Got DSN key from database default_key_1
Got DSN key from database default_key_2
Found 2 keys to try after 0.9 seconds
Decrypting KFX DRM voucher: amzn1.drm-voucher.v1.d6b6caba-0349-42ff-b1aa-fda6a96dcd3d.voucher
KFX DRM voucher successfully decrypted
DeDRM v6.7.0: Ultimately failed to decrypt after 0.9 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 625, in run
  File "calibre_plugins.dedrm.__init__", line 575, in KindleMobiDecrypt
DeDRMError: DeDRM v6.7.0: Ultimately failed to decrypt after 0.9 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Package KFX (from KFX Input) 1.31.0: Packaging C:\Users\AUM\AppData\Local\Temp\calibre_7hyy3d\zxwua4.kfx-zip
Processing container: B07G146556_EBOK.azw
Running file type plugin Package KFX (from KFX Input) failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.kfx_input.package_filetype", line 21, in run
  File "calibre_plugins.kfx_input.package_filetype", line 33, in run_kfx
  File "calibre_plugins.kfx_input.kfxlib.yj_book", line 87, in convert_to_single_kfx
  File "calibre_plugins.kfx_input.kfxlib.yj_book", line 160, in decode_book
  File "calibre_plugins.kfx_input.kfxlib.yj_book", line 267, in get_container
KFXDRMError: Book container B07G146556_EBOK.azw has DRM and cannot be converted
KFX metadata reader (from KFX Input) activated for C:\Users\AUM\AppData\Local\Temp\calibre_7hyy3d\ykqfqa_add_books\0\B07G146556_EBOK.kfx-zip
Added Palestine-Israel ( पॅलेस्टाईन-इस्रायल ) to db in: 2.2
Added 1 books in 5.3 seconds
Worker Launch took: 0.728999853134
Job: 0 Convert book 1 of 1 (Palestine-Israel ( पॅलेस्टाईन-इस्रायल )) finished
Convert book 1 of 1 (Palestine-Israel ( पॅलेस्टाईन-इस्रायल ))
	Conversion options changed from defaults:
	  cover: u'C:\\Users\\AUM\\AppData\\Local\\Temp\\calibre_7hyy3d\\q21ijg.jpeg'
	  read_metadata_from_opf: u'C:\\Users\\AUM\\AppData\\Local\\Temp\\calibre_7hyy3d\\m2fyux.opf'
	  output_profile: u'generic_eink'
	  verbose: 2
	Resolved conversion options
	calibre version: 4.16.0
	{'allow_conversion_with_errors': False,
	 'asciiize': False,
	 'author_sort': None,
	 'authors': None,
	 'base_font_size': 0.0,
	 'book_producer': None,
	 'change_justification': u'original',
	 'chapter': u"//*[((name()='h1' or name()='h2') and re:test(., '\\s*((chapter|book|section|part)\\s+)|((prolog|prologue|epilogue)(\\s+|$))', 'i')) or @class = 'chapter']",
	 'chapter_mark': u'pagebreak',
	 'comments': None,
	 'cover': u'C:\\Users\\AUM\\AppData\\Local\\Temp\\calibre_7hyy3d\\q21ijg.jpeg',
	 'debug_pipeline': None,
	 'dehyphenate': True,
	 'delete_blank_paragraphs': True,
	 'disable_font_rescaling': False,
	 'dont_split_on_page_breaks': False,
	 'duplicate_links_in_toc': False,
	 'embed_all_fonts': False,
	 'embed_font_family': None,
	 'enable_heuristics': False,
	 'epub_flatten': False,
	 'epub_inline_toc': False,
	 'epub_toc_at_end': False,
	 'epub_version': u'2',
	 'expand_css': False,
	 'extra_css': None,
	 'extract_to': None,
	 'filter_css': u'',
	 'fix_indents': True,
	 'flow_size': 260,
	 'font_size_mapping': None,
	 'format_scene_breaks': True,
	 'html_unwrap_factor': 0.4,
	 'input_encoding': None,
	 'input_profile': <calibre.customize.profiles.InputProfile object at 0x05211DF0>,
	 'insert_blank_line': False,
	 'insert_blank_line_size': 0.5,
	 'insert_metadata': False,
	 'isbn': None,
	 'italicize_common_cases': True,
	 'keep_ligatures': False,
	 'language': None,
	 'level1_toc': None,
	 'level2_toc': None,
	 'level3_toc': None,
	 'line_height': 0.0,
	 'linearize_tables': False,
	 'margin_bottom': 5.0,
	 'margin_left': 5.0,
	 'margin_right': 5.0,
	 'margin_top': 5.0,
	 'markup_chapter_headings': True,
	 'max_toc_links': 50,
	 'minimum_line_height': 120.0,
	 'no_chapters_in_toc': False,
	 'no_default_epub_cover': False,
	 'no_inline_navbars': False,
	 'no_svg_cover': False,
	 'output_profile': <calibre.customize.profiles.GenericEink object at 0x05211FF0>,
	 'page_breaks_before': u'/',
	 'prefer_metadata_cover': False,
	 'preserve_cover_aspect_ratio': False,
	 'pretty_print': True,
	 'pubdate': None,
	 'publisher': None,
	 'rating': None,
	 'read_metadata_from_opf': u'C:\\Users\\AUM\\AppData\\Local\\Temp\\calibre_7hyy3d\\m2fyux.opf',
	 'remove_fake_margins': True,
	 'remove_first_image': False,
	 'remove_paragraph_spacing': False,
	 'remove_paragraph_spacing_indent_size': 1.5,
	 'renumber_headings': True,
	 'replace_scene_breaks': u'',
	 'search_replace': '[]',
	 'series': None,
	 'series_index': None,
	 'smarten_punctuation': False,
	 'sr1_replace': None,
	 'sr1_search': None,
	 'sr2_replace': None,
	 'sr2_search': None,
	 'sr3_replace': None,
	 'sr3_search': None,
	 'start_reading_at': None,
	 'subset_embedded_fonts': False,
	 'tags': None,
	 'timestamp': None,
	 'title': None,
	 'title_sort': None,
	 'toc_filter': None,
	 'toc_threshold': 6,
	 'toc_title': None,
	 'transform_css_rules': '[]',
	 'unsmarten_punctuation': False,
	 'unwrap_lines': True,
	 'use_auto_toc': False,
	 'verbose': 2}
	InputFormatPlugin: KFX Input running
	on C:\Users\AUM\AppData\Local\Temp\calibre_7hyy3d\do6moe.kfx-zip
	Software versions: KFX Input 1.31.0, calibre 4.16, Windows-10-10.0.17134-SP0
	KFX Input plugin help is available at https://www.mobileread.com/forums/showthread.php?t=291290
	Converting C:\Users\AUM\AppData\Local\Temp\calibre_7hyy3d\do6moe.kfx-zip
	Processing container: B07G146556_EBOK.azw
	Traceback (most recent call last):
	  File "calibre_plugins.kfx_input.__init__", line 99, in convert
	  File "calibre_plugins.kfx_input.__init__", line 198, in convert_from_kfx
	  File "calibre_plugins.kfx_input.kfxlib.yj_book", line 106, in convert_to_epub
	  File "calibre_plugins.kfx_input.kfxlib.yj_book", line 160, in decode_book
	  File "calibre_plugins.kfx_input.kfxlib.yj_book", line 267, in get_container
	KFXDRMError: Book container B07G146556_EBOK.azw has DRM and cannot be converted
	Python function terminated unexpectedly
	  This book has DRM! (Error Code: 1)
	 Traceback (most recent call last):
	  File "site.py", line 114, in main
	  File "site.py", line 88, in run_entry_point
	  File "site-packages\calibre\utils\ipc\worker.py", line 209, in main
	  File "site-packages\calibre\gui2\convert\gui_conversion.py", line 43, in gui_convert_override
	  File "site-packages\calibre\gui2\convert\gui_conversion.py", line 28, in gui_convert
	  File "site-packages\calibre\ebooks\conversion\plumber.py", line 1110, in run
	  File "site-packages\calibre\customize\conversion.py", line 246, in __call__
	  File "calibre_plugins.kfx_input.__init__", line 104, in convert
	calibre.ebooks.DRMError: This book has DRM!

**Comment by ElleKayEm** [comment]
This book isn't working because it is from Kindle Unlimited.

**Comment by Tushd31** [comment]
> This book isn't working because it is from Kindle Unlimited.

actually , even i thought the same , but that is not the case as i have downloaded more than 200 english novels from kindle unlimited service and converted  , but not getting luck with these regional titles .

**Comment by ElleKayEm** [comment]
That is why this particular book isn't working.  If you have a purchased book of this type, I'll be happy to look at that.  In this case, DeDRM is working as intended.

**Comment by Tushd31** [comment]
Kk sure .Will try to do a purchase and will share the raw file with you .
Thank you so much for this replies .It means alot :)

**Comment by ElleKayEm** [comment]
The file won't do me any good.  Just the log.

**Comment by Tushd31** [comment]
> The file won't do me any good. Just the log.

Kk😅.Sure

**Comment by ElleKayEm** [comment]
Be aware that I don't know if you will be able to remove DRM from a file like that.  At least not until the latest version of DRM is figured out and incorporated into the tools.

**Comment by Tushd31** [comment]
Gotcha .Will have to try as I desperately need some of those titles as they are totally unavailable in any other format than Kindle .
So ,these are worth trying .

**Comment by ElleKayEm** [comment]
You could try with a sample before buying.  I think that would have the same DRM as the full book.

**Comment by ElleKayEm** [comment]
Just found out that the description on the product page may actually say if later versions of Kindle for PC/Mac app are required.  If that is the case, then the DRM can't be removed.
